src/windows/BarcodeScannerProxy.js

```diff
@@ -243,7 +246,19 @@ module.exports = {
             // rotate the preview video
             var videoEncodingProperties = capture.videoDeviceController.getMediaStreamProperties(Windows.Media.Capture.MediaStreamType.videoPreview);
             videoEncodingProperties.properties.insert(ROTATION_KEY, rotDegree);
-            return capture.videoDeviceController.setMediaStreamPropertiesAsync(Windows.Media.Capture.MediaStreamType.videoPreview, videoEncodingProperties);
+
+            try {
+                return capture.videoDeviceController.setMediaStreamPropertiesAsync(Windows.Media.Capture.MediaStreamType.videoPreview, videoEncodingProperties);
+            } catch (e) {
+                if (e.number !== PHOTO_SEQUENCE_IS_IN_PROGRESS) {
+                    throw e;
```

Comment:
There is no intermediate promise variable - so could you please explain how can we reject a returned promise?
I believe it's ok to throw an exception as it will be caught in the topmost promise chain' done.